### PR TITLE
ReMPI: add version 1.2.0

### PR DESCRIPTION
**Pull request by hainest**

### CODE REVIEW

**Review by alalazo — CHANGES_REQUESTED** [review]
There are errors in the build caught by Gitlab pipelines. Did this package, by any chance, gained a dependency on LLVM? It seems one of the error is that `llvm/Pass.h` cannot be found.

**Comment by eugeneswalker** [comment]
Here is an excerpt of the error referenced above. This is from the pipeline job that builds `rempi`:
* https://gitlab.spack.io/spack/spack/-/jobs/1674705

```
...
     1046    libtool: compile:  /home/software/spack/__spack_path_placeholder__
             /__spack_path_placeholder__/__spack_path_placeholder__/__spack_pat
             h_placeholder__/__spack_path_placeholder__/__spack_path_placeholde
             r__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_
             path_placeholder__/__spack_path_placeholder__/__spack_path_placeho
             lder__/__spack_path_placeholder__/__spack_path_placeholder__/__spa
             ck_path_placeholder__/__spack_path_placeholder__/__spack_path_plac
             eholder__/__spack_path_placeholder__/__spack_path_placeholder__/__
             spa/linux-ubuntu18.04-x86_64/gcc-7.5.0/mpich-3.4.2-hxry6lex6h5nsp5
             hbwu7gzu3r4otieun/bin/mpic++ -DHAVE_CONFIG_H -I. -I../.. -Wno-depr
             ecated -lz -lstdc++ -Wno-deprecated -lbfd -rdynamic -g -O2 -MT lib
             reomp_la-reomp_util.lo -MD -MP -MF .deps/libreomp_la-reomp_util.Tp
             o -c reomp_util.cpp -o libreomp_la-reomp_util.o >/dev/null 2>&1
  >> 1047    reomp_ir_pass.cpp:3:10: fatal error: llvm/Pass.h: No such file or 
             directory
     1048     #include <llvm/Pass.h>
     1049              ^~~~~~~~~~~~~
     1050    compilation terminated.
     1051    libtool: compile:  /home/software/spack/__spack_path_placeholder__
             /__spack_path_placeholder__/__spack_path_placeholder__/__spack_pat
             h_placeholder__/__spack_path_placeholder__/__spack_path_placeholde
             r__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_
             path_placeholder__/__spack_path_placeholder__/__spack_path_placeho
             lder__/__spack_path_placeholder__/__spack_path_placeholder__/__spa
             ck_path_placeholder__/__spack_path_placeholder__/__spack_path_plac
             eholder__/__spack_path_placeholder__/__spack_path_placeholder__/__
             spa/linux-ubuntu18.04-x86_64/gcc-7.5.0/mpich-3.4.2-hxry6lex6h5nsp5
             hbwu7gzu3r4otieun/bin/mpic++ -DHAVE_CONFIG_H -I. -I../.. -Wno-depr
             ecated -lz -lstdc++ -Wno-deprecated -g -O2 -MT libreompir_la-reomp
             _util.lo -MD -MP -MF .deps/libreompir_la-reomp_util.Tpo -c reomp_u
             til.cpp -o libreompir_la-reomp_util.o >/dev/null 2>&1
  >> 1052    Makefile:599: recipe for target 'libreompir_la-reomp_ir_pass.lo' f
             ailed
  >> 1053    make[3]: *** [libreompir_la-reomp_ir_pass.lo] Error 1
     1054    make[3]: *** Waiting for unfinished jobs....
     1055    mv -f .deps/libreomp_la-reomp_util.Tpo .deps/libreomp_la-reomp_uti
...
```

**Comment by hainest** [comment]
@alalazo @eugeneswalker I saw these errors yesterday and am working on fixing them. The problem is that the Makefile just shells out to llvm-config instead of using pkgconfig, so I need to do my testing in a sterile environment. I will update when I get a working spack package.

File: var/spack/repos/builtin/packages/rempi/package.py [hunk 1]
```diff
@@ -12,6 +12,7 @@ class Rempi(AutotoolsPackage):
     url      = "https://github.com/PRUNERS/ReMPI/releases/download/v1.0.0/ReMPI-1.0.0.tar.gz"
     tags     = ['e4s']
 
+    version("1.2.0", sha256="130fdb15394ea28be4c06d80c90e654bc91a145e25490477381c3663f2a294f7")
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/rempi/package.py:15]: Confirmed sha256.

**Review by alalazo — CHANGES_REQUESTED** [review]
This issue https://github.com/spack/spack/pull/28640#issuecomment-1024412864 still stands

**Comment by hainest** [comment]
> This issue [#28640 (comment)](https://github.com/spack/spack/pull/28640#issuecomment-1024412864) still stands

I've been working on getting the rest of the ECP Tools SDK working, so this has fallen aside. I'll try to work on it soon.

**Comment by alalazo** [comment]
@hainest Is this PR still being actively worked on?

**Comment by hainest** [comment]
@alalazo I reached out to the reMPI author via email some time ago, but never got a response. Even if I submit a PR to fix the clang issue, I have no idea if/when it would be accepted. For now, reMPI is just broken. We can close this PR until I hear back from them.

**Comment by alalazo** [comment]
Thanks @hainest I'll close, but feel free to ping me as soon as you get a reply.

**Pull request closed without merging**